PhotoScraper/RPI_SIS_PhotoScraper.py

```diff
@@ -46,98 +72,70 @@
-    requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS = 'DES-CBC3-SHA:AES128-SHA:'+requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS
+    # read credentials from (optional) file
+    if len(args.credentials_file) > 0 and os.path.isfile(args.credentials_file):
+        with open(str(args.credentials_file), "r") as f:
+            rcsid = f.readline().strip()
+            passwd = f.readline().strip()
+    else:
+        rcsid = input("RCS ID: ")
+        passwd = getpass.getpass("Password: ")
+
+    print("Setting up selenium", file=sys.stderr)
     driver = webdriver.Chrome(options=chrome_options)
+    driver.implicitly_wait(10)
 
     # open SIS
-    driver.get('https://sis.rpi.edu/')
+    driver.get("https://sis.rpi.edu")
 
-    count = 0
-    while True:
-        count += 1
-        # Types in username & password in login page
-        try:
-            username_box = driver.find_element_by_id('username')
-            break
-        except:
-            if count > 10:
-                print ("ERROR: couldn't find username box")
-                exit(0)
-        # slight delay to allow page to load
-        print ("wait a little longer for the page to load")
-        time.sleep(1)
-    username_box.send_keys(rin_id)
     try:
-        password_box = driver.find_element_by_id('password')
-    except:
-        print ("ERROR: couldn't find password box")
-        exit(0)
-
-    password_box.send_keys(pin_id)
-
-    time.sleep(2)
-
-    try:
-        # click login button
-        login_button = driver.find_element_by_name("submit")
-    except:
-        print ("ERROR: couldn't find submit button")
-        exit(0)
-
-    print ("now we can click login button")
-    login_button.click()
-
-    time.sleep(2)
-
-    while True:
-
-        time.sleep(3)
-
-        if "Rensselaer Self-Service Information System" in driver.page_source:
-            print("success -- made it past duo page")
-            break
-        else:
-            print("please complete duo authentication")
-
-    print ("Continuing with processing...")
-    success = True
-
-    return driver, success
+        print("Logging in", file=sys.stderr)
+        driver.find_element(By.CSS_SELECTOR, 'button[type="submit"]').click()
+        driver.find_element(By.ID, "username").send_keys(rcsid)
+        driver.find_element(By.ID, "password").send_keys(passwd)
+        driver.find_element(By.CSS_SELECTOR, 'button[type="submit"]').click()
+
+        if len(driver.find_elements(By.CSS_SELECTOR, "p.output--error")) > 0:
+            print("Incorrect password", file=sys.stderr)
+            return driver, False
+
+        print("2FA: enter code on your phone", file=sys.stderr)
+        driver.implicitly_wait(25)
+        driver.find_element(By.ID, "dont-trust-browser-button").click()
+        driver.implicitly_wait(10)
+        driver.find_element(
+            By.CSS_SELECTOR, 'a[href="/rss/twbkwbis.P_GenMenu?name=bmenu.P_MainMnu"]'
+        )
+
+        return driver, True
```

Comment:
```suggestion
        try:
            driver.find_element(By.ID, "dont-trust-browser-button").click()
            print("Clicked 'dont-trust-browser-button'", file=sys.stderr)
        except NoSuchElementException:
            pass  # Continue if button is not found
        
        driver.implicitly_wait(10)
        try:
            driver.find_element(By.CSS_SELECTOR, 'a[href="/rss/twbkwbis.P_GenMenu?name=bmenu.P_MainMnu"]')
            return driver, True
        except NoSuchElementException:
            print("Link not found, login failed", file=sys.stderr)
            return driver, False
```

Reply:
We should have something like this to make sure if user clicks either `dont-trust-browser-button` or `trust-browser-button` OR script clicks `dont-trust-browser-button` both would move on to the webpage.

Reply:
This is a good idea but can be implemented with less code, I have taken care of that in a new commit.